### PR TITLE
refactor: GUI: group Frontend responsibilities

### PR DESCRIPTION
**Pull request by allentiak**
Start grouping GUI responsibilities into new package forge.frontend...

Part of #2982.

(See commits for migrated packages.)

### CODE REVIEW

**Comment by tehdiplomat** [comment]
I'm going to run a release this weekend for MOM. Will read through this after that.

File: forge-gui-desktop/src/main/java/forge/itemmanager/filters/AdvancedSearchFilter.java [hunk 1]
```diff
@@ -17,9 +17,9 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.Iterables;
 
+import forge.frontend.components.widgets.IButton;
```
Comment (tehdiplomat) [inline forge-gui-desktop/src/main/java/forge/itemmanager/filters/AdvancedSearchFilter.java:20]: I'm going to run a release this weekend for MOM. Will read through this after that.

Comment (allentiak) [reply forge-gui-desktop/src/main/java/forge/itemmanager/filters/AdvancedSearchFilter.java:20]: Cool!
Just let me know if you need anything.

**Comment by allentiak** [comment]
_(I sent this before, over email, but it seems to have gotten lost somehow.
Anyways, here it is.)_

**********************

As I mentioned in the description, the problem is that there are plenty of classes that mix frontend and backend responsibilities. Eventually, the responsibilities from this last group will have to be (somehow) refactored into the other two.

As this is just the first step into separating those three groups of responsibilities into three clear hierarchies, it is too soon to know the exact final destination for each of those to-be-refactored classes. It could be "game", or it could be somewhere else. This will depend on what the refactored hierarchy ends up looking like.

And yes, by "frontend" I mean "GUI-related" :-)

Have I answered your question?

File: forge-gui-desktop/src/main/java/forge/itemmanager/filters/AdvancedSearchFilter.java (hunk 1 above)
Comment (tehdiplomat) [reply forge-gui-desktop/src/main/java/forge/itemmanager/filters/AdvancedSearchFilter.java:20]: Not sure why this conversation was marked as resolved as I didn't read through this PR yet.

**Comment by allentiak** [comment]
OK. Done.

**Comment by squee1968** [comment]
Could I ask a favor, please? Could you add some gui elements (check boxes) to remember until game ends the choices for things that Forge makes you respond to? Paying costs,  confirming, the tedious stuff.

**Comment by tehdiplomat** [comment]
@squee1968 Feel free to open a feature request, this PR isn't doing anything like that.

**Pull request closed without merging**